pennylane/optimize/gradient_descent.py

```diff
@@ -58,13 +58,30 @@ def step(self, objective_fn, x, grad_fn=None):
                 If ``None``, the gradient function is computed automatically.
 
         Returns:
-            array: the new variable values :math:`x^{(t+1)}`
+            tuple: tuple containing the new variable values :math:`x^{(t+1)}` and the objective
+                function output
         """
 
-        g = self.compute_grad(objective_fn, x, grad_fn=grad_fn)
+        g, forward = self.compute_grad(objective_fn, x, grad_fn=grad_fn)
 
         x_out = self.apply_grad(g, x)
 
+        return x_out, forward
```

Comment:
I was thinking whether we should add
```python
if forward is None:
    forward = objective_fn(x)
```
This accounts for the case when `grad_fn` is already supplied and users have explicitly requested `step_and_cost`. In that case, I'd expect to be getting the cost rather than `None`.
However, this would result in an unnecessary call of the cost function if a user does just `step()` since this calls `step_and_cost()` :thinking: I guess the alternative would be to unchain `step` from `step_and_cost`, but if eventually we plan to converge to just `step_and_cost` then maybe we should have the option of returning `None` when `grad_fn` is supplied :thinking:

Reply:
Maybe it's just fine! Although, perhaps we should mention in the docstring, e.g., on the "Returns" line that users will get `None` if `grad_fn` is supplied.

Reply:
This should be easily doable right? As you say Tom, simply detach `step` and `step_and_cost`, which I am very happy with.

```python
def step_and_cost(self, objective_fn, x, grad_fn=None):
    g, forward = self.compute_grad(objective_fn, x, grad_fn=grad_fn)
    x_out = self.apply_grad(g, x)

    if forward is None:
        forward = objective_fn(x)

    return x_out, forward


def step(self, objective_fn, x, grad_fn=None):
    g, _= self.compute_grad(objective_fn, x, grad_fn=grad_fn)
    x_out = self.apply_grad(g, x)
    return x_out
```

 I have no preference, but we should prioritize intuitive behaviour, rather than worrying about future method convergence.